### PR TITLE
Fix amount JSON object parsing and add unit tests. Rename exceptions.

### PR DESCRIPTION
**Pull request by matejr**
The code throws InvalidJsonException in more cases than before when parsing amount json object. 

SignJson also throws InvalidTransactionException in more cases than before.

### CODE REVIEW

**Comment by sublimator** [comment]
Great, thanks, will hopefully look soon

File: Ripple.Core/Types/Amount.cs [hunk 1]
```diff
@@ -61,13 +62,39 @@ public static Amount FromJson(JToken token)
...
+                    string issuer = null;
+
+                    // we must allow property names that are not in lowercase
+                    foreach(var child in token.Children())
+                    {
+                        var p = child as JProperty;
+                        if (p == null)
+                            throw new InvalidJsonException("Invalid amount object.");
+
+                        if (p.Name.Equals("value", StringComparison.OrdinalIgnoreCase))
```
Comment (sublimator) [inline Ripple.Core/Types/Amount.cs]: This would work with VaLuE eh :) ?

File: Ripple.Core/Types/Amount.cs [hunk 2]
```diff
@@ -61,13 +62,39 @@ public static Amount FromJson(JToken token)
...
+                            value = p.Value.ToString();
+                        else if (p.Name.Equals("currency", StringComparison.OrdinalIgnoreCase))
+                            currency = p.Value.ToString();
+                        else if (p.Name.Equals("issuer", StringComparison.OrdinalIgnoreCase))
+                            issuer = p.Value.ToString();
+                        else
+                            throw new InvalidJsonException($"Unexpected field `{p.Name}`");
```
Comment (sublimator) [inline Ripple.Core/Types/Amount.cs]: `Unexpected field ` vs ` must contain property`

I prefer property for json so as to separate them from the ripple serialisation fields

File: Ripple.Core/Types/Amount.cs (hunk 1 above)
Comment (sublimator) [inline Ripple.Core/Types/Amount.cs]: You want `Value` to work, so it's possible to use the C# naming conventions ?

File: Ripple.Core/Types/Amount.cs (hunk 2 above)
Comment (matejr) [inline Ripple.Core/Types/Amount.cs]: Will fix this.

File: Ripple.Core/Types/Amount.cs (hunk 1 above)
Comment (matejr) [inline Ripple.Core/Types/Amount.cs]: Basically yes. However, I will remove OrdinalIgnoreCase. The [transactions spec](https://ripple.com/build/transactions/) says nothing about case-sensitivity, but apparently the properties are case sensitive.

Comment (sublimator) [inline Ripple.Core/Types/Amount.cs]: Yeah, rippled wouldn't accept Title case, so I think it's best to stick to
that.

**Pull request merged**